### PR TITLE
Slasher min span remove lookback

### LINKED ISSUE #5040: Remove lookback and properly handle min span updates in Slasher
This is an issue for handling min span updates in the slasher, since currently they require going to epoch 0 on a first run, we need a better way to handle them as using the lookback sacrifices security.

following the initial design doc: https://hackmd.prylabs.network/s/H1jO_H9wL

Starting to resolve this issue:

- [x] batch read attestation validators spans from db #5534
- [x] batch write attestation validators spans from db #5534
- [x] use batch read and write to update min span

### PR DESCRIPTION
**Pull request by shayzluf**
Resolves #5040

---

# Description
batch db read and write for min span update in slasher

### CODE REVIEW

File: slasher/detection/attestations/spanner.go [hunk 1]
```diff
@@ -219,13 +234,18 @@ func (s *SpanDetector) updateMinSpan(ctx context.Context, att *ethpb.IndexedAtte
 				indices = append(indices, idx)
 			}
 		}
-		if err := s.slasherDB.SaveEpochSpansMap(ctx, epoch, spanMap); err != nil {
-			return err
-		}
-		if len(indices) == 0 {
-			break
+		//copy(valIndices, indices)
```
Comment (rauljordan) [inline slasher/detection/attestations/spanner.go:237]:
```suggestion
```

File: slasher/detection/attestations/spanner.go [hunk 2]
```diff
@@ -219,13 +234,18 @@ func (s *SpanDetector) updateMinSpan(ctx context.Context, att *ethpb.IndexedAtte
...
+		if !batchRead {
```
Comment (rauljordan) [inline slasher/detection/attestations/spanner.go:238]: There's a lot of complex !batchRead conditionals around this function - can we refactor for easier reading and understanding of the conditionals?

Comment (shayzluf) [reply slasher/detection/attestations/spanner.go:238]: sure

File: slasher/detection/attestations/spanner.go (hunk 1 above)
Comment (shayzluf) [reply slasher/detection/attestations/spanner.go:237]: it should improve performance. commented it out for some reason

**Comment by rauljordan** [comment]
Quick tip @shayzluf if you write `[Resolves] #issuenum` in the PR description it won't resolve the issue, it needs to be `Resolves #issuenum`

**Comment by shayzluf** [comment]
thanks @rauljordan will use it next time

**Pull request merged**